src/main/scala/is/hail/expr/ir/functions/UtilFunctions.scala

```diff
@@ -0,0 +1,30 @@
+package is.hail.expr.ir.functions
+
+import is.hail.asm4s._
+import is.hail.expr.ir._
+import is.hail.expr.types._
+import is.hail.expr.types.coerce
+
+object UtilFunctions extends RegistryFunctions {
+  registerCode[Int]("triangle", TInt32(), TInt32()) { case (_, n: Code[Int]) => n * (n + 1) / 2 }
```

Comment:
I don't think you need the case here, you're taking two arguments, not a pattern match.  That should let you get rid of the [Int] type parameter.

Reply:
hmm... I'm struggling with this a little bit because the `registerCode` definitions take `Code[_]`, so it actually is kind of a pattern match. And the [Int] is the return type so it's not declared anywhere else. I'll see if I can fiddle with the things on that to make things better.

Reply:
I'm still not sure I see the problem.  You should be able to pass Code[X] where Code[_] is expected.

Reply:
If I want to be able to use things like `*` and `+` on a `Code[Int]`, I need to declare that it's a Code[Int] somewhere. The function argument requires `Code[_]`, so specifying that `_` is an `Int` requires the case statement. I'm going to see if I can rewrite it nicer, though.